From 7215e17e5acadf2dd3ecc758882461052c3fc520 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 12 Sep 2021 15:12:51 +0300
Subject: [PATCH] [SW-55766] pci: moving mem access dis/ena to common code

Memory access enable/disable is required by gaudi greco and gaudi2,
and using the same register and same bit.
Moving to common pci code.

Change-Id: I61527cd7fe9ee850a584257850e3bd3f7fbace9b
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi/arm/common.c b/arch/gaudi/arm/common.c
index c836984..08c84e3 100644
--- a/arch/gaudi/arm/common.c
+++ b/arch/gaudi/arm/common.c
@@ -276,32 +276,12 @@
 	hl_pcie_disable_mem_access();
 }
 
-void hl_pcie_disable_mem_access(void)
-{
-	u32 reg;
-
-	/* Disable the PCIe inbound memory requests */
-	reg = regmap_readl(mmPCIE_DBI_STATUS_COMMAND_REG);
-	reg &= ~BIT(1);
-	regmap_writel(reg, mmPCIE_DBI_STATUS_COMMAND_REG);
-}
-
 void hl_pcie_enable_access(void)
 {
 	/* Enable the ELBI access */
 	regmap_writel(0, mmPCIE_CORE_REQ_RETRY_EN);
 
 	hl_pcie_enable_mem_access();
-}
-
-void hl_pcie_enable_mem_access(void)
-{
-	u32 reg;
-
-	/* Enable the PCIe inbound memory requests */
-	reg = regmap_readl(mmPCIE_DBI_STATUS_COMMAND_REG);
-	reg |= BIT(PCIE_DBI_STATUS_COMMAND_REG_PCI_TYPE0_MEM_SPACE_EN_SHIFT);
-	regmap_writel(reg, mmPCIE_DBI_STATUS_COMMAND_REG);
 }
 
 static void preboot_pcie_monitoring(void)
diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index aded8a6..fb5faf5 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -276,28 +276,20 @@
  */
 void hl_pcie_disable_access(void)
 {
-	u32 reg;
-
 	/* Disable the ELBI access */
 	regmap_writel(1, mmPCIE_CORE_REQ_RETRY_EN);
 
 	/* Disable the PCIe inbound memory requests */
-	reg = regmap_readl(mmPCIE_DBI_STATUS_COMMAND_REG);
-	reg &= ~BIT(PCIE_DBI_STATUS_COMMAND_REG_PCI_TYPE0_MEM_SPACE_EN_SHIFT);
-	regmap_writel(reg, mmPCIE_DBI_STATUS_COMMAND_REG);
+	hl_pcie_disable_mem_access();
 }
 
 void hl_pcie_enable_access(void)
 {
-	u32 reg;
-
 	/* Enable the ELBI access */
 	regmap_writel(0, mmPCIE_CORE_REQ_RETRY_EN);
 
 	/* Enable the PCIe inbound memory requests */
-	reg = regmap_readl(mmPCIE_DBI_STATUS_COMMAND_REG);
-	reg |= BIT(PCIE_DBI_STATUS_COMMAND_REG_PCI_TYPE0_MEM_SPACE_EN_SHIFT);
-	regmap_writel(reg, mmPCIE_DBI_STATUS_COMMAND_REG);
+	hl_pcie_enable_mem_access();
 }
 
 void lkd_dev_specific_setup(void)
diff --git a/lib/common/pci.c b/lib/common/pci.c
index 1774105..29590e3 100644
--- a/lib/common/pci.c
+++ b/lib/common/pci.c
@@ -8,6 +8,7 @@
 #include <pcie_dbi_regs.h>
 #include <pcie_aux_regs.h>
 #include <common/pci_ids.h>
+#include <pcie_dbi_masks.h>
 #include <bitops.h>
 #include <preboot.h>
 #include "pci.h"
@@ -192,3 +193,23 @@
 
 	return (dev_id << 16) | PCI_VENDOR_ID_HABANALABS;
 }
+
+void hl_pcie_disable_mem_access(void)
+{
+	u32 reg;
+
+	/* Disable the PCIe inbound memory requests */
+	reg = regmap_readl(mmPCIE_DBI_STATUS_COMMAND_REG);
+	reg &= ~BIT(PCIE_DBI_STATUS_COMMAND_REG_PCI_TYPE0_MEM_SPACE_EN_SHIFT);
+	regmap_writel(reg, mmPCIE_DBI_STATUS_COMMAND_REG);
+}
+
+void hl_pcie_enable_mem_access(void)
+{
+	u32 reg;
+
+	/* Enable the PCIe inbound memory requests */
+	reg = regmap_readl(mmPCIE_DBI_STATUS_COMMAND_REG);
+	reg |= BIT(PCIE_DBI_STATUS_COMMAND_REG_PCI_TYPE0_MEM_SPACE_EN_SHIFT);
+	regmap_writel(reg, mmPCIE_DBI_STATUS_COMMAND_REG);
+}
diff --git a/lib/common/pci.h b/lib/common/pci.h
index 1082a5b..e26eefb 100644
--- a/lib/common/pci.h
+++ b/lib/common/pci.h
@@ -19,6 +19,8 @@
 bool hl_pci_is_mem_enabled(void);
 u32 dev_pci_dbi_cfg_offset_get(void);
 u32 hl_pci_get_ven_id_dev_id(void);
+void hl_pcie_disable_mem_access(void);
+void hl_pcie_enable_mem_access(void);
 
 void hl_pcie_init(void);
 
